### PR TITLE
feat(text editor): Enable use of translation strings for menu items

### PR DESCRIPTION
**Pull request by john-traas**
In order for translations to work some streamlining of the initialisation process of the text editor and to ensure that relevant data is loaded properly was required.

Fixes https://github.com/Lundalogik/crm-feature/issues/4089
Fixes https://github.com/Lundalogik/crm-feature/issues/4118

### CODE REVIEW

**Comment by john-traas** [comment]
@LawrenceBorst I can remove the translations files changes if need be. I didn't vet them with anyone yet and we can make them a part of a separate PR if we have to.

File: src/translations/en.ts [hunk 1]
```diff
@@ -19,4 +19,13 @@ export default {
...
     'file-viewer.open-in-new-tab': 'Open in a new tab',
     'file-viewer.more-actions': 'More…',
+    'editor-menu.bold': 'Bold',
+    'editor-menu.italic': 'Italic',
+    'editor-menu.h1': 'Header level 1',
+    'editor-menu.h2': 'Header level 2',
+    'editor-menu.h3': 'Header level 3',
+    'editor-menu.bulleted-list': 'Bulleted list',
+    'editor-menu.numbered-list': 'Numbered list',
+    'editor-menu.blockquote': 'Blockquote',
```
Comment (LawrenceBorst) [inline src/translations/en.ts:29]: "Block quotation" or "blockquote"?

**Review by LawrenceBorst — COMMENTED** [review]
Looks good and works. Nice refactor in the fourth commit.

If we want to be sure of the languages I think we can only accept Dutch and Swedish. But then I'm pretty sure Norwegian and Danish will look very similar as they do now, so my guess is that they are correct.

File: src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx [hunk 2]
```diff
@@ -113,65 +122,82 @@ export class ProsemirrorAdapter {
...
-        }
+    private getActionBarItems = () => {
+        this.actionBarItems = textEditorMenuItems.map(this.getTranslatedItem);
+    };
 
-        const initialDoc = DOMParser.fromSchema(mySchema).parse(
-            initialContentElement,
-        );
+    private getTranslatedItem = (item) => {
+        const newItem = cloneDeep(item);
```
[the remark this answers is not in the document]
Comment (john-traas) [reply src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx:139]: It was indeed 😅  - Adrian helped me track that one down.

File: src/translations/en.ts (hunk 1 above)
Comment (john-traas) [reply src/translations/en.ts:29]: Blockquote is correct

File: src/components/text-editor/text-editor.tsx [hunk 3]
```diff
@@ -33,6 +34,13 @@ export class TextEditor implements FormComponent<string> {
     @Prop()
     public contentType: 'markdown' | 'html' = 'markdown';
 
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
```
[the remark this answers is not in the document]
Comment (john-traas) [reply src/components/text-editor/text-editor.tsx]: Updated.

File: src/translations/en.ts [hunk 4]
```diff
@@ -19,4 +19,13 @@ export default {
...
+    'editor-menu.link': 'Add Link',
```
[the remark this answers is not in the document]
Comment (john-traas) [reply src/translations/en.ts]: Updated to 'Link'

File: src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx [hunk 5]
```diff
@@ -53,6 +57,13 @@ export class ProsemirrorAdapter {
     @Prop()
     public value: string;
 
+    /**
+     * Defines the language for translations.
+     * Will translate the translatable strings on the components.
+     */
```
[the remark this answers is not in the document]
Comment (john-traas) [reply src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx]: Updated

File: src/translations/en.ts [hunk 6]
```diff
@@ -19,4 +19,13 @@ export default {
     'file-viewer.open-in-fullscreen': 'Open in fullscreen',
     'file-viewer.open-in-new-tab': 'Open in a new tab',
     'file-viewer.more-actions': 'More…',
+    'editor-menu.bold': 'Bold',
+    'editor-menu.italic': 'Italic',
+    'editor-menu.h1': 'Header level 1',
+    'editor-menu.h2': 'Header level 2',
+    'editor-menu.h3': 'Header level 3',
```
Comment (Kiarokh) [inline src/translations/en.ts:26]: Not a big deal, but in MS Words, Apple Pages, etc… they have always just called these "Heading 1", "Heading 2", etc… I'm a fan of "less is more". So if you wanna change the, I'd be very happy

File: src/translations/en.ts (hunk 1 above)
Comment (Kiarokh) [inline src/translations/en.ts:29]: Don't we need `'editor-menu.strikethrough': 'Strikethrough',` as well

Comment (john-traas) [reply src/translations/en.ts:29]: We don't have strike through as part of our menu. We can add it, but it wasn't in the options to be added to the menu when I built the new one.

File: src/translations/en.ts (hunk 6 above)
Comment (john-traas) [reply src/translations/en.ts:26]: That's no problem, we can change that

File: src/components/text-editor/text-editor.tsx [hunk 7]
```diff
@@ -33,6 +34,13 @@ export class TextEditor implements FormComponent<string> {
...
+    @Prop({ reflect: true })
+    public language: Languages;
```
[the remark this answers is not in the document]
Comment (john-traas) [reply src/components/text-editor/text-editor.tsx:41]: We have the default English in the original menu items which was why I didn't add that here. But I can add it.

[the remark this answers is not in the document]
Comment (john-traas) [reply src/components/text-editor/text-editor.tsx]: Updated now.

File: src/translations/en.ts (hunk 6 above)
Comment (john-traas) [reply src/translations/en.ts]: Updated

Comment (LawrenceBorst) [reply src/translations/en.ts]: `Header` or `Heading` for this? Currently we have committed `Header`.

Although this is probably not so important, but I really wanted to resolve this comment 😅

Comment (john-traas) [reply src/translations/en.ts]: The commit with the change was missed. I've rebased and it's all there now 😅

**Review by jgroth — APPROVED** [review]

**Pull request merged**